### PR TITLE
UCS/MEMTRACK: Revamp memtrack service to avoid changing allocation size

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6925/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6927/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6929/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6930/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@alinask pls take a look

File: src/ucs/debug/memtrack.h [hunk 1]
```diff
@@ -26,27 +25,19 @@ enum {
...
-    char                  name[UCS_MEMTRACK_NAME_MAX];
-    size_t                size;
-    size_t                peak_size;
-    size_t                count;
-    size_t                peak_count;
-    ucs_memtrack_entry_t  *next;
-};
+typedef struct ucs_memtrack_entry {
+    size_t                  size;
+    size_t                  peak_size;
```
Comment (alinask) [inline src/ucs/debug/memtrack.h]: could you please document these fields? at least peak_size and peak_count.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6934/ for details (Mellanox internal link).

File: src/ucs/debug/memtrack.c [hunk 2]
```diff
@@ -68,358 +55,229 @@ static ucs_stats_class_t ucs_memtrack_stats_class = {
...
+    /* remote pointer from hash */
+    entry = kh_val(&ucs_memtrack_context.ptrs, iter).entry;
+    size  = kh_val(&ucs_memtrack_context.ptrs, iter).size;
+    kh_del(ucs_memtrack_ptr_hash, &ucs_memtrack_context.ptrs, iter);
 
-    /* Update total size */
-    ucs_assert(entry->size >= buffer->size);
-    entry->size -= buffer->size;
+    /* update counts */
+    ucs_memtrack_entry_update(entry, -size);
```
Comment (alinask) [inline src/ucs/debug/memtrack.c]: why negative size?

**Comment by yosefe** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/6941/ for details (Mellanox internal link).

**Pull request merged**